llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -41371,8 +41371,18 @@ bool X86TargetLowering::SimplifyDemandedVectorEltsForTargetNode(
     case X86ISD::UNPCKH:
     case X86ISD::BLENDI:
       // Integer ops.
+    case X86ISD::PCMPEQ:
```

Comment:
Don't know enough about x86 but seems like there would need to be more tests hitting these cases?

Reply:
I split this into 2 commits so the review can see the x86 regressions and the followup that addresses them - I intend to squash+merge these - do you think that's enough?